Add Prompting 101 Tile to independent-ai-topics

kelsey-steven-ada opened the pull request:
[Addresses this Asana ticket](https://app.asana.com/0/1201700438643002/1205889394913675/f)

I'd particularly love feedback around the challenge questions and if they seem useful =]

kelsey-steven-ada committed:
Add placeholders for Prompting 101 lessons and assets. Update config to include new Prompting 101 tile. Add main lesson content for Making Requests From ChatGPT (WIP, missing screenshots & text for setting up ChatGPT)

kelsey-steven-ada committed:
Add screenshots and step by step instructions for signing up for ChatGPT to the Prompting 101 tile lesson

kelsey-steven-ada committed:
Add content and prompt examples to the propmting 101 tile's lesson 'Improving ChatGPT's Responses'

kelsey-steven-ada committed:
Update some examples to show off more use cases in 'Improving ChatGPT's Responses' and add a check for understanding question

kelsey-steven-ada committed:
Add check for understanding to Prompting 101 - Making Requests From ChatGPT

kelsey-steven-ada committed:
Fill out main content for prompting 101 lesson 3 'Verifying Information from ChatGPT'

kelsey-steven-ada committed:
Add check for understanding to lesson 3 of Prompting 101, update UUID to unique value for all understanding checks

kelsey-steven-ada committed:
Rename second tile and description to better fit contents

kelsey-steven-ada committed:
Change all proformatted code blocks over to quote blocks for readability. Remove a repeated paragraph about miscategorizing sentiment from prompting 101 - lesson 2

kelsey-steven-ada committed:
Clarify statement about generating non-existent citations in lesson 3 of prompting 101

kelsey-steven-ada committed:
Fix grammar in lesson 3 at start of paragraph about when are generative tools not useful

char-adadev commented on 02-prompting-101/making-requests-from-chatgpt.md:
```diff
@@ -0,0 +1,204 @@
+
+The readings from the Generative AI Overview lesson outlined a handful of uses for Large Language Models in the business world, but how can they help us, right now, in our own lives? We’ve listed some more use cases below with examples of how folks have used them, but there are a couple caveats to think about before we dive into our own prompting:
+- Like any website with unknown sources, if an AI tool like ChatGPT is generating concrete information like dates and names that we did not provide it, we want to do our own web searches to confirm those details are accurate and not fabricated or otherwise false. We’ll dive more into techniques for verifying responses shortly.
+- There are legal considerations if we use AI tools for public work or businesses. Many tools have been trained on massive data sets with text, art and other works scraped from the internet without legal clearance. Unless a tool has been built solely with data that has been given permission to be used for training and generation, it may not be legal to use the art or text generated by an AI tool  ...
```
The last sentence was a bit difficult for me to parse the first time reading it. What do you think of this small change?

```suggestion
- There are legal considerations if we use AI tools for public work or businesses. Many tools have been trained on massive data sets with text, art and other works scraped from the internet without legal clearance. Using the art or text generated by an AI tool verbatim may be legally restricted unless the tool has been exclusively constructed with data that has been authorized for training and generation.
```

kelsey-steven-ada replied:
That reads much clearer to me too, thanks!

char-adadev commented on 02-prompting-101/making-requests-from-chatgpt.md:
```diff
@@ -0,0 +1,204 @@
+
+### Signing up
+
+1. Our first step is to visit [chat.openai.com](chat.openai.com) and press the "Sign Up" button to start creating an account. 
```
```suggestion
1. Our first step is to visit [chat.openai.com](https://chat.openai.com) and press the "Sign Up" button to start creating an account. 
```

char-adadev commented on 02-prompting-101/making-requests-from-chatgpt.md:
```diff
@@ -0,0 +1,204 @@
+![Email verification message](./assets/verify-email-message.png)  
+*Fig. Email verification message*
+
+5. You can log in with your new account at [chat.openai.com](chat.openai.com) once your email is verified. The first time you log in, you'll be asked to enter your first name, last name, and birth date to verify your age.
```
```suggestion
5. You can log in with your new account at [chat.openai.com](https://chat.openai.com) once your email is verified. The first time you log in, you'll be asked to enter your first name, last name, and birth date to verify your age.
```

char-adadev commented on 02-prompting-101/making-requests-from-chatgpt.md:
```diff
@@ -0,0 +1,204 @@
+![Submit verification code](./assets/phone-verification-code.png)   
+*Fig. Form to submit verification code*
+
+8. At this point you should be redirected a brand new chat page! We should see "ChatGPT 3.5" at the top, letting us know we are working with OpenAI's 3.5 model. 
```
```suggestion
8. At this point you should be redirected to a brand new chat page! We should see "ChatGPT 3.5" at the top, letting us know we are working with OpenAI's 3.5 model. 
```

char-adadev commented on 02-prompting-101/making-requests-from-chatgpt.md:
```diff
@@ -0,0 +1,204 @@
+![Regenerate viewing options](./assets/regenerate-viewing-tools.png)  
+*Fig. response tools after generating a second response*
+
+The first number indicates how many versions of the response there are, and the second number is which version of the response we are currently viewing. The arrows are buttons that let us navigate forwards and backwards through the responses so we can see all of the versions.
```
```suggestion
The second number indicates how many versions of the response there are, and the first number is which version of the response we are currently viewing. The arrows are buttons that let us navigate forwards and backwards through the responses so we can see all of the versions.

For example, if we generate a response 3 times, when the indicator shows `< 1 / 3 >`, this means we are viewing the first output of the prompt, `< 2 / 3 >` means we are viewing the second output of the prompt, and so on.
```

kelsey-steven-ada replied:
Thanks for catching the number/position swap here, the added example is really helpful. 🙏🏻

char-adadev commented on 02-prompting-101/improving-chatGPTs-responses.md:
```diff
@@ -0,0 +1,518 @@
+
+### !end-callout
+
+This incorrect answer is clearly not desired, but we can often get to a correct response for this question by asking the AI tool to show it's work and reason itself to a correct answer. Try out this updated prompt in either the same chat or a new one:
```
```suggestion
This incorrect answer is clearly not desired, but we can often get to a correct response for this question by asking the AI tool to show its work and reason itself to a correct answer. Try out this updated prompt in either the same chat or a new one:
```

char-adadev commented on 02-prompting-101/improving-chatGPTs-responses.md:
```diff
@@ -0,0 +1,518 @@
+> Sentiment: negative  
+> Themes: UI, cursor  
+
+If we include no examples, no instructions, and simply submit our list of feedback as a prompt, generative AI tools like ChatGPT will still try to provide a helpful response, likely about how to address these pieces of user feedback. 
```
I was a bit lost reading this section because I wasn't entirely sure what the prompt to Chat was - is it literally the list of feedback with no other context?

kelsey-steven-ada replied:
That's exactly what's happening, the prompt is literally just the feedback. Would it be helpful to include a sentence before we show the example around "a prompt can be just data without instructions"?

kelsey-steven-ada replied:
Edit: I added the line as a suggestion on the PR so it's easier to see/review.

I'm thinking above the feedback when introducing the example, we could start the paragraph with something like:

> We're going to look at a prompt that's different from others we've seen – this time we're not going to include _any_ direct instructions on what the generative AI should do. Let's say that we moderate a forum...

char-adadev commented on 02-prompting-101/verifying-information-from-chatGPT.md:
```diff
@@ -0,0 +1,150 @@
+
+### Looking for SMART resources
+
+When doing our own searching, we also need to be asking ourselves if the sources we’re using to verify information are trustworthy. Anyone can publish anything on the internet, but there are frameworks we can use to ask ourselves questions about a website or source to help us investigate if it may be worthwhile. Once such framework is the SMART Check:
```
```suggestion
When doing our own searching, we also need to be asking ourselves if the sources we’re using to verify information are trustworthy. Anyone can publish anything on the internet, but there are frameworks we can use to ask ourselves questions about a website or source to help us investigate if it may be worthwhile. One such framework is the SMART Check:
```

char-adadev commented on 02-prompting-101/verifying-information-from-chatGPT.md:
```diff
@@ -0,0 +1,150 @@
+
+## Summary 
+
+Generative AI tools are very powerful, but at this stage, there are several issues around hallucinations, the cutoff date of their training data, inability connect to the internet to gather new or further information, and inability to share sources from their training data, which together mean that we cannot trust any facts that they state without verification. 
```
```suggestion
Generative AI tools are very powerful, but at this stage, there are several issues around hallucinations, the cutoff date of their training data, inability to connect to the internet to gather new or further information, and inability to share sources from their training data, which together mean that we cannot trust any facts that they state without verification. 
```

char-adadev commented on 02-prompting-101/verifying-information-from-chatGPT.md:
```diff
@@ -0,0 +1,150 @@
+##### !end-question
+
+##### !options
+* S
```
```suggestion
* S - Scope
```

char-adadev commented on 02-prompting-101/verifying-information-from-chatGPT.md:
```diff
@@ -0,0 +1,150 @@
+
+##### !options
+* S
+* M
```
```suggestion
* M - Motivation
```

char-adadev commented on 02-prompting-101/verifying-information-from-chatGPT.md:
```diff
@@ -0,0 +1,150 @@
+##### !options
+* S
+* M
+* A
```
```suggestion
* A - Authority & Accuracy
```

char-adadev commented on 02-prompting-101/verifying-information-from-chatGPT.md:
```diff
@@ -0,0 +1,150 @@
+* S
+* M
+* A
+* R
```
```suggestion
* R - Relevancy & Reliability
```

char-adadev commented on 02-prompting-101/verifying-information-from-chatGPT.md:
```diff
@@ -0,0 +1,150 @@
+* M
+* A
+* R
+* T
```
```suggestion
* T - Timely
```

char-adadev reviewed (approved):
Amazingggg work on this, Kelsey!! Reading through this makes me super excited about getting this in front of students. It's gonna be really cool to see the various ways that folx will utilize this tool.

I left a few suggestions; nothing major jumped out at me. Thank you so much!

kelsey-steven-ada committed:
Update 02-prompting-101/making-requests-from-chatgpt.md

Update wording of legal considerations for use.

Co-authored-by: char <110567531+char-adadev@users.noreply.github.com>

kelsey-steven-ada committed:
Update 02-prompting-101/making-requests-from-chatgpt.md

Add missing `https://` to front of URL.

Co-authored-by: char <110567531+char-adadev@users.noreply.github.com>

kelsey-steven-ada committed:
Update 02-prompting-101/making-requests-from-chatgpt.md

Add missing `https://` to front of URL.

Co-authored-by: char <110567531+char-adadev@users.noreply.github.com>

kelsey-steven-ada commented on 02-prompting-101/verifying-information-from-chatGPT.md:
```diff
@@ -0,0 +1,150 @@
+##### !end-options
+
+##### !answer
+* T
```
```suggestion
* T - Timely
```

kelsey-steven-ada committed:
Apply suggestions from code review to prompting 101 lesson

Small grammar/spelling fixes, adding examples or further description for clarity.

Co-authored-by: char <110567531+char-adadev@users.noreply.github.com>

kelsey-steven-ada commented on 02-prompting-101/improving-chatGPTs-responses.md:
```diff
@@ -0,0 +1,518 @@
+
+Especially when you need the output formatted a particular way or in a way that is difficult to describe, we can improve an AI Tool’s response by including examples (called “Shots” when talking about prompt engineering) of what it is that we want the AI tool to do. When we include a single example, this is called “Single Shot” prompting, and when we give it 2 or more examples it gets called “Few  ...
+
+Let's say that we moderate a forum and want to categorize feedback from our users. We're going start from a zero-shot prompt and iterate on it until we have a few-shot prompt. For our example, we have a list of feedback like:
```
```suggestion
We're going to look at a prompt that's different from others we've seen – this time we're not going to include any instructions telling the generative AI what it should do. Let's say that we moderate a forum and want to categorize feedback from our users. We're going start from a zero-shot prompt and iterate on it until we have a few-shot prompt. For our example, we have a list of feedback like:
```

kelsey-steven-ada commented on 02-prompting-101/improving-chatGPTs-responses.md:
```diff
@@ -0,0 +1,518 @@
+> Sentiment: negative  
+> Themes: UI, cursor  
+
+If we include no examples, no instructions, and simply submit our list of feedback as a prompt, generative AI tools like ChatGPT will still try to provide a helpful response, likely about how to address these pieces of user feedback. 
```
```suggestion
If we include no examples, no instructions, and submit our list of feedback itself as the prompt, generative AI tools like ChatGPT will still try to provide a helpful response, likely about how to address these pieces of user feedback. 
```

kelsey-steven-ada committed:
Clarify zero shot prompting example
